Source/gg2/Objects/Weapons/Rifle.events/User Event 3.xml

```diff
@@ -48,7 +48,8 @@ while(len&gt;1) {
         if(collision_line(x1,y1,xm,ym,Obstacle,true,true)&gt;=0) {
             hitline = true;
         } else if (collision_line(x1,y1,xm,ym,Character,true,true)&gt;=0) {
-            hitline = true;
+            if (collision_line(x1,y1,xm,ym,Character,true,true).team != team)
```

Comment:
I'm pretty sure this is buggy. If you have two characters on the same line segment that is being checked, `collision_line` will only return one of them. If one is an enemy and one a teammate, there is a 50/50 chance the shot will also pierce the enemy because the collision code will never see it. Fixing this is not trivial.